src/compiler.ts

```diff
@@ -5445,7 +5444,9 @@ export class Compiler extends DiagnosticEmitter {
         }
       }
     } else {
-      body.push(this.compileStatement(bodyStatement));
+      body.push(bodyStatement instanceof Expression
```

Comment:
This should probably be `bodyStatement.kind != NodeKind.BLOCK`, as `instanceof` isn't supported (yet, without a GC).